### PR TITLE
Issue 1130 - native uris should be valid

### PR DESCRIPTION
**Pull request by philkcw**
### Summary
Fixes https://github.com/doorkeeper-gem/doorkeeper/issues/1130

### CODE REVIEW

**Comment by philkcw** [comment]
Sigh.  How can code coverage decrease when I added one line and one test?

**Comment by ThibG** [comment]
This is missing changes to the test file:
https://github.com/doorkeeper-gem/doorkeeper/pull/1060/files#diff-b2085d14cb8374956712da8c8d55e8ef

**Comment by nbulaj** [comment]
The tests are red. Also could you please squash all the commits to a single one once it finished? Thanks!

**Comment by philkcw** [comment]
@ryansch I tried to add in your other change but it broke your tests and my tests as well, so I reverted back to the one-line change that I'm most interested in. 😄

**Comment by philkcw** [comment]
@ryansch @thibaudgg I'll take another look at the changes you mentioned this weekend.

**Comment by thibaudgg** [comment]
@philkcw I think you meant @ThibG and not me :)

**Comment by ryansch** [comment]
Ha! Fair enough!

**Comment by Gargron** [comment]
I just want to tell you both good luck. We're all counting on you.

**Comment by philkcw** [comment]
@ryansch I added in your change.  @ThibG I added in your tests.  Maybe I should have cherry-picked instead?  @nbulaj the build passes and I've squashed.  Any more comments?

File: spec/models/doorkeeper/application_spec.rb [hunk 1]
```diff
@@ -266,6 +266,23 @@ module Doorkeeper
         let(:confidential) { false }
         it { expect(subject).to eq(false) }
       end
+
+      context 'when the application does not support confidentiality' do
```
Comment (philkcw) [inline spec/models/doorkeeper/application_spec.rb]: I added this to get the code coverage to pass.

**Comment by nbulaj** [comment]
Looks good to me too, thanks!

**Pull request merged**